doc/source/development/rolling_operations_with_numba.rst

```diff
@@ -0,0 +1,234 @@
+Rolling Operations with Numba
+=============================
+
+Summary
+-------
+
+This proof of concept (POC) demonstrates using `Numba <http://numba.pydata.org/>`_ instead of ``Cython``
+to compute ``rolling.mean`` and ``rolling.apply`` without introducing any user facing API changes.
+The benefits of using ``Numba`` include:
+
+#. Performance parity or improvement over ``Cython``
+#. Eliminate shipping C-extensions
+#. Ease of debugging and maintenance as pure Python code
+
+Additionally this POC includes an new API that allows users to create custom window boundary calculations
```

Comment:
Add a sentence explaining *why* we would want to create custom window boundaries. this allows the user to easily overide window calculcations, for example to account for holidays or missing periods, w/o requiring changes in pandas itself.